joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -999,6 +996,13 @@ controller_interface::CallbackReturn JointTrajectoryController::on_activate(
     state_desired_ = state;
     last_commanded_state_ = state;
   }
+  else
+  {
+    // Initialize current state storage from hardware
+    read_state_from_hardware(state_current_);
+    read_state_from_hardware(state_desired_);
+    read_state_from_hardware(last_commanded_state_);
```

Comment:
When we are already fixing this. I would propose to rename the method to `read_state_from_state_interfaces` – it is a longer but clearer name. Read state from hardware is a bit unclear. Especially now with having chaining of state interfaces, this might be confusing.

Reply:
fine!